### PR TITLE
awsvpc docs

### PR DESCRIPTION
**Pull request by bboreham**
Relating to #2341

### CODE REVIEW

File: site/installing-weave.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ With Weave Net downloaded onto your VMs or hosts, you are ready to launch a Weav
 
 CoreOS users see [here](/guides/networking-docker-containers-with-weave-on-coreos/) for an example of installing Weave using cloud-config.
 
+Amazon AWS users may want to use AWSVPC?
```
Comment (awh) [inline site/installing-weave.md]: Did you mean to leave this in here?

**Comment by awh** [comment]
Looks good :+1:

File: site/installing-weave.md (hunk 1 above)
Comment (bboreham) [inline site/installing-weave.md]: I was pondering whether this is the right place to link from - the others are Guides.

**Comment by rade** [comment]
A diagram would be useful.

**Comment by bboreham** [comment]
> A diagram would be useful.

The analogous diagram for FDP is in a separate "How it works" doc; do we want that?

**Comment by rade** [comment]
> "How it works" doc; do we want that?

No.

**Comment by bboreham** [comment]
Added diagram.

**Comment by bboreham** [comment]
Ping @abuehrle for doc review

File: site/installing-weave.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ With Weave Net downloaded onto your VMs or hosts, you are ready to launch a Weav
 
 CoreOS users see [here](/guides/networking-docker-containers-with-weave-on-coreos/) for an example of installing Weave using cloud-config.
 
+Amazon EC2 users may want to use [AWS-VPC mode](/site/using-weave/awsvpc.md)
```
Comment (rade) [inline site/installing-weave.md]: I am concerned that EC2 users will jump there as a matter of course. The AWS-VPC mode is a fair bit harder to set up than sleeve/fastdp, so, IMO we should steer users towards the latter first and only recommend AWS-VPC for users who reach performance limits.

Also, this is missing a full stop.

File: site/using-weave/awsvpc.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+allocated by Amazon.
+
+###Using AWS-VPC mode
+
+Launch Weave Net, adding the `--awsvpc` flag:
+
+    $ weave launch --awsvpc [other hosts]
+
+You still need to supply the names or IP addresses of other hosts in
+your cluster.
```
Comment (rade) [inline site/using-weave/awsvpc.md]: How can one tell whether awsvpc mode is being used by a node? Is there some indication in `weave status`? `weave status connections`?

File: site/using-weave/awsvpc.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+###Limitations
+
+- AWS-VPC mode does not inter-operate with other Weave Net modes; it
+  is all or nothing.  In this mode, all hosts in a cluster must be AWS
+  instances. We hope to ease this limitation in future.
+- The AWS network does not support multicast.
+- The number of hosts in a cluster is limited by the maximum size of
+  your AWS route table.  This is limited to 50 entries though you
+  can request an increase to 100 by contacting Amazon.
```
Comment (rade) [inline site/using-weave/awsvpc.md]: Presumably that limit can also be exceeded with fewer hosts, due to fragmentation. Worth mentioning?

Related, when the limit is reached, what happens? And how can a user tell that is the root cause for whatever stops working?

File: site/using-weave/awsvpc.md (hunk 3 above)
Comment (bboreham) [inline site/using-weave/awsvpc.md]: Nothing in `weave status` at present.

`weave report` has `"OverlayDiagnostics": { "awsvpc": null }`

    $ weave status connections
    -> 172.31.23.45:6783     established awsvpc 26:a1:90:ca:f0:13(ip-172-31-23-45)

Comment (rade) [inline site/using-weave/awsvpc.md]: The `weave status connections` docs need updating, and should be referenced here.

Comment (bboreham) [inline site/using-weave/awsvpc.md]: I looked at those, and the text still seems correct - it doesn't say what connection types may be printed.

Comment (rade) [inline site/using-weave/awsvpc.md]: it's documented elsewhere - https://www.weave.works/docs/net/latest/using-weave/fastdp/ "Viewing Connection Mode".

Should really be moved. So we can then point to it from here (and the fastdp docs).

File: site/using-weave/awsvpc.md (hunk 4 above)
Comment (bboreham) [inline site/using-weave/awsvpc.md]: I believe what happens in the current code is errors in the log file.

I captured this in https://github.com/weaveworks/weave/issues/2322#issuecomment-224629128

File: site/installing-weave.md (hunk 2 above)
Comment (bboreham) [inline site/installing-weave.md]: I reworded the text surrounding the link to awsvpc docs

File: site/using-weave/awsvpc.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- AWS-VPC mode does not inter-operate with other Weave Net modes; it
+  is all or nothing.  In this mode, all hosts in a cluster must be AWS
+  instances. (We hope to ease this limitation in future.)
+- The AWS network does not support multicast.
+- The number of hosts in a cluster is limited by the maximum size of
+  your AWS route table.  This is limited to 50 entries though you
+  can request an increase to 100 by contacting Amazon.
+- All of your containers must be on the same network, with no subnet
+  isolation. (We hope to ease this limitation in future.)
+
```
Comment (brb) [inline site/using-weave/awsvpc.md]: Should we mention that `weave reset` called by the last alive peer does not remove its routes?

Comment (bboreham) [inline site/using-weave/awsvpc.md]: That issue was fixed in #2356, so no.

File: site/using-weave/awsvpc.md [hunk 6]
```diff
@@ -19,9 +19,14 @@ on which they live.
...
-existing role, then extend it or create a new role with a policy
-that allows the necessary actions:
+table via its
+[IAM Role](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html).
+If you have an existing IAM Role then extend it, otherwise create a
+new role. The role must have a
+[policy](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-policies-for-amazon-ec2.html)
+attached which allows the following
+[actions](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Operations.html):
+
```
Comment (abuehrle) [inline site/using-weave/awsvpc.md]: LGTM

File: site/installing-weave.md [hunk 7]
```diff
@@ -32,11 +32,15 @@ With Weave Net downloaded onto your VMs or hosts, you are ready to launch a Weav
 
 CoreOS users see [here](/guides/networking-docker-containers-with-weave-on-coreos/) for an example of installing Weave using cloud-config.
 
-Amazon EC2 users may want to use [AWS-VPC mode](/site/using-weave/awsvpc.md)
-
 Amazon ECS users see [here](https://github.com/weaveworks/integrations/blob/master/aws/ecs/README.md)
 for the latest Weave AMIs and [here](http://weave.works/guides/service-discovery-with-weave-aws-ecs.html) to get started with Weave Net on ECS.
 
+On Amazon EC2 the standard installation instructions at the top of
+this page provide the simplest set-up and most flexibility, but there
```
Comment (abuehrle) [inline site/installing-weave.md]: If you're on Amazon EC2, the standard installation instructions at the top of this page, provide the simplest setup and the most flexibility.  A [special no-overlay mode for EC2](/site/using-weave/awsvpc.md) can optionally be enabled, which allows containers to communicate at the full speed of the underlying network.

**Pull request merged**